crates/ruff_linter/src/rules/ruff/rules/missing_fstring_syntax.rs

```diff
@@ -88,8 +88,10 @@ fn should_be_fstring(
         return false;
     }
 
+    let fstring_expr = format!("f{}", locator.slice(literal.range()));
+
     let Ok(ast::Expr::FString(ast::ExprFString { value, .. })) =
-        parse_expression(&format!("f{}", locator.slice(literal.range())))
+        parse_expression(&fstring_expr)
```

Comment:
I believe there's a parse_expression_at method in the parser that gives you absolute offsets (less confusing)

Reply:
Oh great! That would definitely make things simpler 😁

Reply:
Actually, wait - I think it's a bit more complicated since putting the `f` at the front changes the resulting offset. We could just add 1 to the start offset (that was my original approach, actually) but I think slicing from the parsed string would be simpler and more immediately understandable.

Reply:
We've used code like `"f".text_len()` in multiple places but I guess this is fine as well.

Reply:
Ah good point. I missed the fact that we parse a modified string. We could subtract `-1` from the start but that would fail apart if the string is at the very start of the file (not unlikely for docstrings). 

I think it's fine. It might be worth adding a comment somewhere mentioning that all offsets are relative to `fstring_expr`